src/momento/auth/credential_provider.py

```diff
@@ -42,6 +46,11 @@ def from_environment_variable(
         Returns:
             CredentialProvider
         """
+        warn(
+            "from_environment_variable is deprecated, use from_environment_variables_v2 instead",
+            DeprecationWarning,
+            stacklevel=2,
+        )
```

Comment:
We'll want to update the docstrings on the methods as well, ie write that they are deprecated.